### PR TITLE
[fix][Transaction] Fix transaction admin redirect get 500 due to getCause

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
Transaction admin `getCoordinatorInternalStats`  add a getCause when try catch.
Which make redirect fail.

* Exception is thrown by `validateTopicOwnership`, sync() call get() in 
  `sync(()-> validateTopicOwnershipAsync(topicName, authoritative));`
* Exception already got cause by in `validateTopicOwnershipAsync`
### Modifications
Delete get cause if not ExecutionException

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java [hunk 1]
```diff
@@ -504,7 +504,11 @@ protected void internalGetCoordinatorInternalStats(AsyncResponse asyncResponse,
                         "This Broker is not configured with transactionCoordinatorEnabled=true."));
             }
         } catch (Exception e) {
-            asyncResponse.resume(new RestException(e.getCause()));
+            if (e instanceof ExecutionException) {
+                asyncResponse.resume(new RestException(e.getCause()));
+            } else {
+                asyncResponse.resume(new RestException(e));
+            }
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java:511]: How about using `resumeAsyncResponseExceptionally` ?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+@Slf4j
+public class AdminApiTransactionMultiBrokerTest extends TransactionTestBase {
+
+    private static final int NUM_BROKERS = 16;
+    private static final int NUM_PARTITIONS = 16;
+    
+    @BeforeMethod
+    protected void setup() throws Exception {
+        setUpBase(NUM_BROKERS, NUM_PARTITIONS, NAMESPACE1 + "/test", 0);
```
Comment (gaoran10) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: Maybe we need to set the configuration `allowAutoTopicCreation` to `false`, according to the description in the issue https://github.com/apache/pulsar/issues/14790 the `allowAutoTopicCreation` is `false`.

Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: I think that was used to test another question. allowAutoTopicCreation is false, has nothing to do with this PR, and we use admin in setupBase to create the required topic.

Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: Does this PR is used to fix issue #14790? If so, maybe we need to create the same environment for the issue.

Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: When we don't know what the problem is, we need to set up the same environment to reproduce the problem. And when we fix the problem, we only need to focus on the problem itself

Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: I can start a cluster of brokers with the same configuration as the issue to test this modification. If there is no problem, I don't think we need to include extraneous variables in CI

Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: And this is not fix the issue https://github.com/apache/pulsar/issues/14790

Comment (gaoran10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: OK. It seems that this PR is linked with issue #14790 before.

Comment (liangyepianzhou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionMultiBrokerTest.java:44]: My fault, did not read this issue carefully

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java [hunk 3]
```diff
@@ -501,7 +501,11 @@ protected void internalGetCoordinatorInternalStats(AsyncResponse asyncResponse,
                         "This Broker is not configured with transactionCoordinatorEnabled=true."));
             }
         } catch (Exception e) {
-            asyncResponse.resume(new RestException(e.getCause()));
+            if (e instanceof ExecutionException) {
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java:504]: Please check the method `resumeAsyncResponseExceptionally`, seems that it already handles the `ExecutionException` case, we don't need to check it here.

**Review by Technoboy- — APPROVED** [review]

**Review by congbobo184 — APPROVED** [review]

**Pull request merged**